src/main/scala/com/microsoft/hyperspace/actions/CreateAction.scala

```diff
@@ -65,9 +65,20 @@ class CreateAction(
   }
 
   private def isValidIndexSchema(config: IndexConfig, schema: StructType): Boolean = {
+    // First we flatten the schema. Instead of having struct of leaves
+    // the flatten method will return a list of field names.
+    // The second step is escaping the field names as Parquet does not works well
```

Comment:
Could you elaborate on this "Parquet does not work well"?

Reply:
Sorry,  it is a bit misleading and not properly formulated. It is about this bug on `partitionBy` in Apache Spark: https://issues.apache.org/jira/browse/SPARK-18084.

I try writing a better comment.